### PR TITLE
Adds support for survey mode

### LINKED ISSUE #625: Adding survey mode
A survey-type question should be introduced in frontend editor.

### PR DESCRIPTION
**Pull request by dalmia**
Fixes #625 

## Summary

Survey mode feature is added.

## Test Plan

- [x] Test Responsiveness
   - [ ] Laptop (1200px)
   - [ ] Tablet (760px)
   - [ ] Phone (320px)
- [x] Cross-Browser Testing
   - [ ] Chrome
   - [ ] Firefox
   - [ ] Safari (use BrowserStack)
- [x] Local Language Support
- [x] Hygiene and Housekeeping
   - [ ] Self-review
   - [ ] Comments have been added appropriately
   - [ ] ~Check for bundle size [here](https://bundlephobia.com/) if adding a package~
   - [ ] Added relevant details like Labels/Projects/Milestones etc.
   - [ ] ~If adding or removing any environment variable:~
       - [ ] update `docs/ENV.md`
       - [ ] update Github Workflow files
       - [ ] update DockerFile
       - [ ] update the secrets for staging and production
- [ ] Testing
   - [x] Wrote tests
   - [x] Tested locally
   - [ ] Tested on staging
   - [ ] Tested on an actual physical phone
   - [ ] Tested on production
- [ ] ~Lighthouse Checks~
   - [ ] Images have `alt` attributes
   - [ ] Any `` tags have `width` and `height` specified
   - [ ] Any `target="_blank"` links have `rel="noopener"`
   - [ ] Only SVGs are used as images. If PNGs are used, their size has been optimised.
   - [ ] Any SVGs without text have their `aria-label` attributes set

### CODE REVIEW

File: tests/unit/components/Editor/ItemEditor.spec.js [hunk 1]
```diff
@@ -417,6 +439,18 @@ describe("ItemEditor.vue", () => {
       ).toBeTruthy();
     });
 
+    it("check for survey mode", async () => {
```
Comment (dalmia) [inline tests/unit/components/Editor/ItemEditor.spec.js:442]:
```suggestion
    it("hides warning for survey mode", async () => {
```

File: tests/unit/components/Editor/ItemEditor.spec.js [hunk 2]
```diff
@@ -417,6 +439,18 @@ describe("ItemEditor.vue", () => {
...
+      await wrapper
+        .find('[data-test="surveyQuestionCheckbox"]')
+        .setChecked("checked");
+      await flushPromises();
+      expect(
+        wrapper.find('[data-test="subjectiveQuestionContainer"]').exists()
+      ).toBeTruthy();
+      expect(wrapper.find('[data-test="options"]').exists()).toBeFalsy();
```
Comment (dalmia) [inline tests/unit/components/Editor/ItemEditor.spec.js:450]: this should already not be shown irrespective of survey mode or not right? not needed here?

File: tests/unit/components/Editor/ItemEditor.spec.js [hunk 3]
```diff
@@ -417,6 +439,18 @@ describe("ItemEditor.vue", () => {
...
     });
 
+    it("check for survey mode", async () => {
+      await wrapper
+        .find('[data-test="surveyQuestionCheckbox"]')
+        .setChecked("checked");
+      await flushPromises();
+      expect(
+        wrapper.find('[data-test="subjectiveQuestionContainer"]').exists()
+      ).toBeTruthy();
```
Comment (dalmia) [inline tests/unit/components/Editor/ItemEditor.spec.js:449]: upon making survey_mode `True` this should not exist right?

File: src/components/Editor/ItemEditor.vue [hunk 4]
```diff
@@ -164,34 +189,35 @@
         class="p-2"
         data-test="subjectiveQuestionContainer"
       >
-        <!-- checkbox -->
+        <!-- checkbox for setting max char limit -->
         <label class="inline-flex items-center mt-3">
           <input
             type="checkbox"
-            class="form-checkbox h-5 w-5 text-primary focus:ring-transparent"
+            style="box-shadow: none"
```
Comment (dalmia) [inline src/components/Editor/ItemEditor.vue:196]: can't you use `shadow-none`

File: src/components/Editor/ItemEditor.vue [hunk 5]
```diff
@@ -85,6 +85,30 @@
...
+          />
+          <span class="ml-2 text-gray-700 text-sm">{{
+            $t("editor.item_editor.survey_mode")
+          }}</span>
+        </label>
+        <inline-svg
+          :src="getImageSource('question-circle-regular.svg')"
+          class="w-5 h-5 pl-1 items-center place-self-center"
+          v-tooltip="{
+            content: surveyModeTooltip,
```
Comment (dalmia) [inline src/components/Editor/ItemEditor.vue:106]:
```suggestion
            content: $t('tooltip.editor.item_editor.survey_mode'),
```

File: src/components/Editor/ItemEditor.vue [hunk 6]
```diff
@@ -510,6 +540,9 @@ export default {
   },
 
   computed: {
+    surveyModeTooltip() {
+      return this.$t("tooltip.editor.item_editor.survey_mode");
+    },
```
Comment (dalmia) [inline src/components/Editor/ItemEditor.vue:545]: no need to define this separately - can directly add the tooltip text in the template itself

File: tests/unit/components/Editor/ItemEditor.spec.js [hunk 7]
```diff
@@ -417,6 +439,14 @@ describe("ItemEditor.vue", () => {
       ).toBeTruthy();
     });
 
+    it("hides warning for survey mode", async () => {
```
Comment (dalmia) [inline tests/unit/components/Editor/ItemEditor.spec.js:442]:
```suggestion
    it("toggles survey mode", async () => {
```

File: tests/unit/pages/Editor.spec.js [hunk 8]
```diff
@@ -1047,6 +1047,42 @@ describe("Editor.vue", () => {
     expect(wrapper.vm.itemImage).toBe(imageURL);
   });
 
+  it("checks whether subjective warning is shown when survey question", async () => {
+    const isSurveyQuestion = jest.spyOn(Editor.computed, "isSurveyQuestion");
+    wrapper = mount(Editor);
+
+    const dummyItemDetailsWithSurvey = clonedeep(global.dummyItemDetails);
```
Comment (dalmia) [inline tests/unit/pages/Editor.spec.js:1054]: this is not being used

File: tests/unit/pages/Editor.spec.js [hunk 9]
```diff
@@ -1047,6 +1047,42 @@ describe("Editor.vue", () => {
...
+
+    await wrapper.setData({
+      items: clonedeep(global.dummyItems),
+      itemDetails: clonedeep(global.dummyItemDetails),
+      currentItemIndex: 2,
+    });
+
+    expect(isSurveyQuestion).toHaveBeenCalled();
```
Comment (dalmia) [inline tests/unit/pages/Editor.spec.js:1062]: `toHaveBeenCalled` check is applied for methods and not for computed properties.

File: tests/unit/pages/Editor.spec.js [hunk 10]
```diff
@@ -1047,6 +1047,42 @@ describe("Editor.vue", () => {
...
+
+    const dummyItemDetailsWithSurvey = clonedeep(global.dummyItemDetails);
+
+    await wrapper.setData({
+      items: clonedeep(global.dummyItems),
+      itemDetails: clonedeep(global.dummyItemDetails),
+      currentItemIndex: 3,
+    });
+
+    expect(isSurveyQuestion).toHaveBeenCalled();
```
Comment (dalmia) [inline tests/unit/pages/Editor.spec.js:1080]: `toHaveBeenCalled` check is applied for methods and not for computed properties.

File: tests/unit/pages/Editor.spec.js [hunk 11]
```diff
@@ -1047,6 +1047,42 @@ describe("Editor.vue", () => {
...
+    expect(
+      wrapper.find('[data-test="subjectiveWarning"]').exists()
+    ).toBeFalsy();
+  });
+
+  it("checks whether subjective warning is shown when not a survey question", async () => {
+    const isSurveyQuestion = jest.spyOn(Editor.computed, "isSurveyQuestion");
+    wrapper = mount(Editor);
+
+    const dummyItemDetailsWithSurvey = clonedeep(global.dummyItemDetails);
```
Comment (dalmia) [inline tests/unit/pages/Editor.spec.js:1072]: not used

File: src/components/Items/Question/Footer.vue [hunk 12]
```diff
@@ -97,6 +98,10 @@ export default {
       default: "",
       type: String,
     },
+    isSelectedItemSurveyQuestion: {
```
Comment (dalmia) [inline src/components/Items/Question/Footer.vue:101]:
```suggestion
    isSurveyQuestion: {
```

File: src/pages/Embeds/Plio.vue [hunk 13]
```diff
@@ -652,7 +677,10 @@ export default {
           ? (this.numCorrect += 1)
           : (this.numWrong += 1);
         this.numSkipped -= 1;
-      } else if (this.isSubjectiveQuestionAnswered(itemIndex, userAnswer)) {
+      } else if (
+        this.isSubjectiveQuestionAnswered(itemIndex, userAnswer) &&
+        userAnswer != null
```
Comment (dalmia) [inline src/pages/Embeds/Plio.vue:682]: `isSubjectiveQuestionAnswered` being `true` automatically means `userAnswer != null` right?

File: src/components/Editor/ItemEditor.vue (hunk 4 above)
Comment (Salonichoudhary07) [reply src/components/Editor/ItemEditor.vue:196]: no this will not work

**Pull request merged**